src/main/java/io/jenkins/pluginhealth/scoring/service/UpdateCenterService.java

```diff
@@ -36,18 +36,26 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UpdateCenterService {
     private final ObjectMapper objectMapper;
     private final String updateCenterURL;
+    private final PluginService pluginService;
 
-    public UpdateCenterService(@Value("${jenkins.update.center}") String updateCenterURL) {
+    public UpdateCenterService(@Value("${jenkins.update.center}") String updateCenterURL, PluginService pluginService) {
+        this.pluginService = pluginService;
         this.objectMapper = Jackson2ObjectMapperBuilder.json().build();
         this.updateCenterURL = updateCenterURL;
     }
 
+    @Scheduled(cron = "${cronexpression}", zone = "UTC")
+    public void updateDatabase() throws IOException {
```

Comment:
This is the newly introduced method and since I was using `pluginService` variable I had to initialise it in the constructor, which in turn told me about the problems in the UpdateCenterServiceTest.java file.

There I had to pass an object of `PluginService` class while creating an object of `UpdateCenterService` class.

Reply:
I'm not sure it's write to have this in this service.